src/kitovu/sync/filecache.py

```diff
@@ -0,0 +1,70 @@
+"""FileCache keeps track of the state of the files, remotely and locally. It exists to determine if the local file
+has been changed between two synchronisation processes and allows for a conflict handling accordingly."""
+
+from enum import Enum
+import json
+import pathlib
+import typing
+
+from kitovu.sync import syncplugin
+from kitovu import utils
+
+# sync-process:
+# every time we sync something, the FileCache gets read, upon which kitovu decides what needs to be downloaded.
+# The files get downloaded, for each file, the FileCache gets updated for the files that were synchronised.
+
+
+class Filestate(Enum):
+    """Used to discern in which places files have changed"""
+    LOCAL = 1
+    REMOTE = 2
+    BOTH = 3
+    NONE = 4
+
+
+class File:
+    def __init__(self, local_digest_at_synctime: str,
+                 relative_path_with_filename: pathlib.PurePath,
+                 plugin: syncplugin.AbstractSyncPlugin) -> None:
+        file_properties = {"path": relative_path_with_filename,
+                           "plugin": plugin,
+                           "digests": {
+                                "cached digest": local_digest_at_synctime,
+                                }
+                           }
+
+
+class FileCache:
+    FILECACHE: str = "filecache.json"
+
+    def write_file_to_filecache(self, file: File):
+        file_to_json = json.dumps(file)
+        try:
+            with self.FILECACHE.open("a+") as f:  # a+ = if it exsists, append; if not, create a new one
+                f.write(file_to_json)
+        except OSError as error:
+            raise utils.UsageError(f"Could not write the file, {error}")
```

Comment:
I wouldn't add error handling for this right now - it's very unlikely to happen (the file cache should always be writable), and a proper exception makes things much easier to debug than a "Could not write the file" message out of context (what file?). We can still discuss whether this kind of thing is needed at all, but for a first iteration, don't worry about it.